pandas/sparse/array.py

```diff
@@ -322,31 +332,51 @@ def take(self, indices, axis=0):
             # return scalar
             return self[indices]
 
-        indices = np.atleast_1d(np.asarray(indices, dtype=int))
-
-        # allow -1 to indicate missing values
+        indices = com._ensure_platform_int(indices)
```

Comment:
yeh, this is a function that (for now), put with the ``.take`` functions in ``core/common.py`` (though I am going to move them I think to ``algorithms.py`` (so can put there in anticipation).